WC_Pagantis.php

```diff
@@ -314,10 +315,38 @@ public function pagantisActivation()
         update_option('woocommerce_pagantis_settings', $settings);
     }
 
+    /**
+     * Product simulator PP o SELECTABLE
+     */
+    public function pagantisAddProductSimulatorPP()
+    {
+        $simType = strtolower($this->extraConfig['PAGANTIS_SIMULATOR_DISPLAY_TYPE']);
+        $validTypes = array('sdk.simulator.types.selectable_text_custom','sdk.simulator.types.product_page');
+        if (in_array($simType, $validTypes)) {
+            return $this->pagantisAddProductSimulator();
+        }
+
+        return false;
+    }
+
+    /**
+     * Product simulator
+     */
+    public function pagantisAddProductSimulatorSP()
+    {
+        $simType = strtolower($this->extraConfig['PAGANTIS_SIMULATOR_DISPLAY_TYPE']);
+        $validTypes = array('sdk.simulator.types.selectable_text_custom','sdk.simulator.types.product_page');
+        if (!in_array($simType, $validTypes)) {
+            return $this->pagantisAddProductSimulator();
+        }
+
+        return false;
+    }
+
     /**
      * Product simulator
      */
-    public function pagantisAddProductSimulator()
+    private function pagantisAddProductSimulator()
```

Comment:
This method name is not very easy to read or understand, in my opinion a better name would be addProductSimulatoAfterCartForm()

Reply:
This in the main method, it's called from pagantisAddProductSimulatorAfterCartButton or pagantisAddProductSimulatorAfterPrice, according the action